api/comments/serializers.py

```diff
@@ -0,0 +1,162 @@
+from datetime import datetime
+from rest_framework import serializers as ser
+from framework.guid.model import Guid
+from framework.auth.core import Auth
+from website.project.model import Node, Comment
+from rest_framework.exceptions import ValidationError, PermissionDenied, NotFound
+from api.base.exceptions import InvalidModelValueError
+from api.base.utils import absolute_reverse
+from api.base.serializers import (JSONAPISerializer,
+                                  JSONAPIHyperlinkedRelatedField,
+                                  JSONAPIHyperlinkedGuidRelatedField,
+                                  JSONAPIHyperlinkedIdentityField,
+                                  IDField, TypeField, LinksField,
+                                  AuthorizedCharField)
+
+
+class CommentReport(object):
+    def __init__(self, user_id, category, text):
+        self._id = user_id
+        self.category = category
+        self.text = text
+
+
+class CommentSerializer(JSONAPISerializer):
+
+    filterable_fields = frozenset([
+        'deleted'
+    ])
+
+    id = IDField(source='_id', read_only=True)
+    type = TypeField()
+    content = AuthorizedCharField(source='return_content')
+
+    user = JSONAPIHyperlinkedRelatedField(view_name='users:user-detail', lookup_field='pk', lookup_url_kwarg='user_id', link_type='related', read_only=True)
+    node = JSONAPIHyperlinkedRelatedField(view_name='nodes:node-detail', lookup_field='pk', lookup_url_kwarg='node_id', link_type='related', read_only=True)
+    target = JSONAPIHyperlinkedGuidRelatedField(link_type='related', meta={'type': 'get_target_type'})
+    replies = JSONAPIHyperlinkedIdentityField(view_name='comments:comment-replies', lookup_field='pk', link_type='self', lookup_url_kwarg='comment_id')
+    reports = JSONAPIHyperlinkedIdentityField(view_name='comments:comment-reports', lookup_field='pk', lookup_url_kwarg='comment_id', link_type='related', read_only=True)
+
+    date_created = ser.DateTimeField(read_only=True)
+    date_modified = ser.DateTimeField(read_only=True)
+    modified = ser.BooleanField(read_only=True, default=False)
```

Comment:
Having a default value in the comment model seems reasonable. (And since None and False are both Falsy, the impact should be minimal :european_post_office: )

I agree that you'll need to change the serializer either way, though, since there will be old comments in the DB. We've had to do this for other serializers so it should be pretty small.